Brokerages/WebSocketWrapper.cs

```diff
@@ -82,18 +167,27 @@ public void Connect()
         /// </summary>
         public void Close()
         {
-            _wrapped.Close();
+            try
+            {
+                _wrapped.CloseAsync(WebSocketCloseStatus.NormalClosure, string.Empty, _cancellationTokenSource.Token);
+
+                OnClose(new WebSocketCloseData(WebSocketCloseStatus.NormalClosure, string.Empty));
+            }
+            catch (Exception exception)
+            {
+                OnError(new WebSocketError(exception.Message, exception));
+            }
         }
 
         /// <summary>
-        /// Wraps IsAlive
+        /// Returns true if the WebSocketState is Open
         /// </summary>
-        public bool IsOpen => _wrapped.IsAlive;
+        public bool IsOpen => _wrapped.State == WebSocketState.Open;
```

Comment:
`_wrapped == null` before `Initialize` is called, perhaps use the null navigation operation:  `?.`